workflows/nanoseq.nf

```diff
@@ -0,0 +1,456 @@
+            ch_software_versions = ch_software_versions.mix(ALIGN_MINIMAP2.out.samtools_version.first().ifEmpty(null))
+            ch_samtools_multiqc  = ALIGN_MINIMAP2.out.ch_sortbam_stats_multiqc.ifEmpty([])
+        } else {
+
+            /*
+             * SUBWORKFLOW: Align fastq files with graphmap2 and sort bam files
+             */
+            ALIGN_GRAPHMAP2 ( ch_fasta_index, ch_fastq )
+            ch_view_sortbam = ALIGN_GRAPHMAP2.out.ch_sortbam
+            ch_software_versions = ch_software_versions.mix(ALIGN_GRAPHMAP2.out.graphmap2_version.first().ifEmpty(null))
+            ch_software_versions = ch_software_versions.mix(ALIGN_GRAPHMAP2.out.samtools_version.first().ifEmpty(null))
+            ch_samtools_multiqc  = ALIGN_GRAPHMAP2.out.ch_sortbam_stats_multiqc.ifEmpty([])
+        }
+
+        ch_bedtools_version = Channel.empty()
+        if (!params.skip_bigwig){
+
+            /*
+             * SUBWORKFLOW: Convert BAM -> BEDGraph -> BigWig
+             */
+            BEDTOOLS_UCSC_BIGWIG ( ch_view_sortbam )
+            ch_bedtools_version = ch_bedtools_version.mix(BEDTOOLS_UCSC_BIGWIG.out.bedtools_version.first().ifEmpty(null))
+            ch_software_versions = ch_software_versions.mix(BEDTOOLS_UCSC_BIGWIG.out.bedgraphtobigwig_version.first().ifEmpty(null))
+        }
+        if (!params.skip_bigbed){
+
+            /*
+             * SUBWORKFLOW: Convert BAM -> BED12 -> BigBED
+             */
+            BEDTOOLS_UCSC_BIGBED ( ch_view_sortbam )
+            ch_bedtools_version = ch_bedtools_version.mix(BEDTOOLS_UCSC_BIGBED.out.bedtools_version.first().ifEmpty(null))
+            ch_software_versions = ch_software_versions.mix(BEDTOOLS_UCSC_BIGBED.out.bed12tobigbed_version.first().ifEmpty(null))
+        }
+        ch_software_versions = ch_software_versions.mix(ch_bedtools_version.first().ifEmpty(null))
+
+        ch_view_sortbam
+            .map { it -> [ it[0], it[3] ] }
+            .set { ch_sortbam }
+    } else {
+        ch_sample
+            .map { it -> if (it[6].toString().endsWith('.bam')) [ it[0], it[6] ] }
+            .set { ch_sample_bam }
+        BAM_RENAME ( ch_sample_bam )
+        ch_sortbam = BAM_RENAME.out.sortbam_quant
+    }
+
+    ch_featurecounts_gene_multiqc       = Channel.empty()
+    ch_featurecounts_transcript_multiqc = Channel.empty()
+    if (!params.skip_quantification && (params.protocol == 'cDNA' || params.protocol == 'directRNA')) {
+
+        // Check that reference genome and annotation are the same for all samples if perfoming quantification
+        // Check if we have replicates and multiple conditions in the input samplesheet
+        REPLICATES_EXIST    = false
+        MULTIPLE_CONDITIONS = false
+        ch_sample.map{ it[2] }.unique().toList().set { fastas }
+        ch_sample.map{ it[3] }.unique().toList().set { gtfs }
+        // BUG: ".val" halts the pipeline ///////////////////////
```

Comment:
Does this error still apply?

Reply:
These no longer apply. I'm keeping them there as a note.

Reply:
OK :+1: